cpp/src/io/utilities/parsing_utils.cuh

```diff
@@ -691,12 +648,15 @@ struct ConvertFunctor {
                                                       parse_options_view const& opts,
                                                       bool as_hex)
   {
-    if constexpr (cudf::is_timestamp<T>() or cudf::is_duration<T>()) {
-      static_cast<T*>(out_buffer)[row] = decode_value<T>(begin, end, opts);
-      return true;
+    // TODO what's invalid input
+    if constexpr (cudf::is_timestamp<T>()) {
+      static_cast<T*>(out_buffer)[row] = to_timestamp<T>(begin, end, opts.dayfirst);
```

Comment:
It looks like this is where the timestamp/duration overloads for `decode_value` would have hit before, so now it's just inlining that logic. Is that right?

Reply:
Yes. It seemed like another level of indirection only for timestamp, duration types. Rest of the types had SFINAE anyway. So, used `if constexpr` directly here.

Reply:
moved to  #12022